wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Dropdown/Simple.js

```diff
@@ -513,62 +519,83 @@ define(
+				var activeIndex = listItems.indexOf(activeItem);
+				if (EventKey.End(event) || EventKey.Home(event)) {
+					activeIndex = -1;
+				}
+				
+				for (var i = activeIndex + 1; i < listItems.length; i++) {
+					if (isValidItem(listItems[i])) {
+						newActiveItem = listItems[i];
+						break;
 					}
-					var activeIndex = listItems.indexOf(activeItem);
-					var newActiveItem = null;
-					
-					var isValidItem = function(listItem) {
-						return !listItem.classList.contains('dropdownDivider') && listItem.clientHeight > 0;
-					};
-					
-					for (var i = activeIndex + 1; i < listItems.length; i++) {
+				}
+				
+				if (newActiveItem === null) {
+					for (i = 0; i < listItems.length; i++) {
 						if (isValidItem(listItems[i])) {
 							newActiveItem = listItems[i];
 							break;
 						}
 					}
-					
-					if (newActiveItem === null) {
-						for (i = 0; i < listItems.length; i++) {
-							if (isValidItem(listItems[i])) {
-								newActiveItem = listItems[i];
-								break;
-							}
-						}
-					}
-					newActiveItem.focus();
 				}
+				
+				newActiveItem.focus();
 			}
 			else if (EventKey.Enter(event) || EventKey.Space(event)) {
 				event.preventDefault();
-				var activeItem = document.activeElement;
-				if (activeItem.nodeName === 'LI') {
-					var target = activeItem;
-					if (target.childElementCount === 1 && (target.children[0].nodeName === 'SPAN' || target.children[0].nodeName === 'A')) {
-						target = target.children[0];
-					}
+				
+				var target = activeItem;
+				if (target.childElementCount === 1 && (target.children[0].nodeName === 'SPAN' || target.children[0].nodeName === 'A')) {
+					target = target.children[0];
+				}
+				
+				dropdown = _dropdowns.get(_activeTargetId);
+				button = elBySel('.dropdownToggle', dropdown);
+				require(['Core'], function(Core) {
+					var mouseEvent = elData(dropdown, 'a11y-mouse-event') || 'click';
+					Core.triggerEvent(target, mouseEvent);
 					
-					var dropdown = _dropdowns.get(_activeTargetId);
-					var button = elBySel('.dropdownToggle', dropdown);
-					target.click();
 					if (button) button.focus();
-				}
+				});
 			}
-			else if (EventKey.Tab(event)) {
+			else if (EventKey.Escape(event) || EventKey.Tab(event)) {
 				event.preventDefault();
-				var dropdown = _dropdowns.get(_activeTargetId);
-				var button = elBySel('.dropdownToggle', dropdown);
+				
+				dropdown = _dropdowns.get(_activeTargetId);
+				button = elBySel('.dropdownToggle', dropdown);
+				// Remote controlled drop-down menus may not have a dedicated toggle button, instead the
+				// `dropdown` element itself is the button.
+				if (button === null && !dropdown.classList.contains('dropdown')) {
+					button = dropdown;
+				}
+				
 				this._toggle(null, _activeTargetId);
 				if (button) button.focus();
 			}
+		},
+		
+		_getNextVisibleItem: function(activeItem, index) {
+			var items = Array.prototype.slice.call(elBySelAll('li', activeItem.closest('.dropdownMenu')));
```

Comment:
This method does not look correct to me.

Reply:
It's a leftover method, I'll remove it, thanks.